### ISSUE TITLE
Integrate QR code scanning for the users joining the queue

### ISSUE THREAD

**Issue by sajmalyousef** [SimplQ/simplQ-frontend#471]
**Is your feature request related to a problem? Please describe.**
Currently we generate QR codes for the users who create queues, however inorder to join the queue users have to rely on other apps to scan 

**Describe the solution you'd like**
A QR code scanning  option at the homepage, similar to the one below

![image](https://user-images.githubusercontent.com/13188906/104810808-3fa4e880-581d-11eb-9d02-9204aa0b851c.png)

**Additional context**
Need to check if this is feasible

**Comment by daltonfury42** [comment]
Now that we generate QR codes, I feel this is a good addition. That way people who don't have (or don't know if they have) a QR scanner can be covered. I could find two libraries which can do this:

https://www.npmjs.com/package/react-qr-reader
https://www.npmjs.com/package/react-qr-scanner

The first one has more stats (weekly downloads and stars on github), but it has not been updated for 2 years. The second looks like an actively maintained fork.

1. Shall we do a material style floating action button at the bottom middle of the screen, as in your screenshot?

![image](https://user-images.githubusercontent.com/11256376/104813318-7b947980-582e-11eb-944d-5d1bf74d2620.png)

2. I am thinking we should disable this on desktop (as it won't be used on desktop) or should we keep it? On the printed barcode, we can ask people to "Visit simplq.me and scan this barcode to get into the queue".

**Comment by maaverik** [comment]
I agree that this is a good addition, but I'm not keen on using a floating button. It would make things inconsistent considering the create and join buttons aren't floating and even if we consider join will be used much more than the create button, we would need to have two floating buttons for join or just one which gives further options to join by name or QR (having two floating buttons feels too bloaty). A third button would anyway need to be added to the CreateJoin form, right?

And I agree with not having this option on desktops.

**Comment by daltonfury42** [comment]
I was thinking of not adding a third button on the CreateJoin form. Thought of floating because then we don't need much of a change... Three buttons in a row is a bit cluttered, in my opinion.

**Comment by maaverik** [comment]
I think we should make a design prototype first and compare to see which one looks cleaner and more consistent. I agree it would be less cluttered, but won't look consistent anymore.

**Comment by daltonfury42** [comment]
This is a quick approximation of what I had in mind, maybe we could use a different color. Can you come up with something quick for what you have in mind?


https://user-images.githubusercontent.com/11256376/104838563-b6a3b500-58e1-11eb-93fe-9379387152b0.mov

**Comment by maaverik** [comment]
It just looks a little weird to me because it's makes it seem like joining with a QR is the most common action on the site while I feel like joining with a queue name would be just as common for people to do, if not more.  Still, a floating button is a good idea on mobile. What do you think about a floating button that just says join and on clicking, it scrolls to the top where we have all 3 buttons?

![Screenshot from 2021-01-17 23-47-46](https://user-images.githubusercontent.com/12583528/104852048-7f9fc480-591e-11eb-8db1-8caf7cdea2ed.png)

**Comment by daltonfury42** [comment]
In my view, joining by QR, sharing a link to the queue etc are just fancy add-ons that are not going to be used much. This is just my hypothesis, but I think 90% of the people will be added manually by the queue manager. That will be the primary method of getting people into the queue, if SimplQ gets used by a business.

Take for example a Rasta Cafe or a busy Truffles, or a hospital. People can always get in on their on, there is anyway a need of a person at the gate, and the person is better off greeting and quickly adding the person. 

TLDR; These according to me are just fancy features that can get a prospective client to start using the app, but mostly won't be used in real world.

***

So in that sense, I would want the fancy feature to stand out when someone is browsing the home page. It doesn't add real value.

**Comment by maaverik** [comment]
That's fair. My issue with this is just with the UI being inconsistent cause of this change. Either way, actual implementation isn't very hard it seems. Those libraries mentioned above should work fairly easily.

**Comment by prabureddy** [comment]
In my view, join by QR will be less used when compared to share using social media. This feature adds look to our website but not used much. So, let it be we can add this feature.

**Comment by daltonfury42** [comment]
Yeah. It adds look, but more than that, it boasts of a cool capability, even if nobody would ever use it.

I think if a real business adopts SimplQ, they would mostly greet customers and add them manually. 🤞

**Comment by micaiah-effiong** [comment]
Hi, if this feature is still wanted by the team, I'd like to work on it.

**Comment by maaverik** [comment]
Hi @micaiah-effiong. Yes, this is still open. Do you have some ideas on the design? We hadn't finalized how it should look, so if you have any ideas, and if you could post a basic figma design or just a screenshot to start with, that'd be great.

**Comment by micaiah-effiong** [comment]
I considered the scan icon could come up as part of the mobile navigation bar, since it is small and makes for good accessibility.

I am not strong on my design skills, but I can come up with something.

**Comment by daltonfury42** [comment]
Scanning QR to enter the queue is more of a good selling point "feature", when talking to potential customers. 

I'd prefer a floating material style QR button, but this is something we could decide on later. So, you can implement as per your comfort, navbar is fine, and then we can fix/move the button positioning and style etc in a follow up PR.

**Comment by micaiah-effiong** [comment]
If so the I'll go with this design https://github.com/SimplQ/simplQ-frontend/issues/471#issuecomment-761778902.
